node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -136,14 +136,16 @@ object Main {
       case _                                 => conf.printHelp()
     }
 
-    program.doOnFinish(
-      _ =>
-        Task.delay {
-          proposeService.close()
-          replService.close()
-          deployService.close()
-        } >> console.close
-    )
+    Task.delay(
+      sys.addShutdownHook(
+        (for {
+          _ <- Task.delay(proposeService.close())
+          _ <- Task.delay(replService.close())
+          _ <- Task.delay(deployService.close())
+          _ <- console.close
+        } yield ()).unsafeRunSync(scheduler)
+      )
+    ) >> program
```

Comment:
are the above delays really needed?

Isn't it the same as:
```scala
    Task.delay(
      sys.addShutdownHook {
        proposeService.close()
        replService.close()
        deployService.close()
        console.close.unsafeRunSync(scheduler)
      }
    ) >> program
```

Reply:
Yeah, you are right. Fixed.